metrology/reporter/statsd.py

```diff
@@ -51,40 +55,66 @@ def socket(self):
 
     def write(self):
         for name, metric in self.registry:
+
+            send = functools.partial(self.send_metric,
+                                     name=name, metric=metric)
+
             if isinstance(metric, Meter):
-                self.send_metric(name, 'meter', metric, 'count')
+                send(mtype='meter',
+                     keys=['count', 'one_minute_rate', 'five_minute_rate',
+                           'fifteen_minute_rate', 'mean_rate'])
             if isinstance(metric, Gauge):
-                self.send_metric(name, 'gauge', metric, 'value')
+                send(mtype='gauge', keys=['value'])
             if isinstance(metric, UtilizationTimer):
-                self.send_metric(name, 'timer', metric, 'mean')
+                send(mtype='timer',
+                     keys=['count', 'one_minute_rate', 'five_minute_rate',
+                          'fifteen_minute_rate', 'mean_rate', 'min', 'max',
+                          'mean', 'stddev', 'one_minute_utilization',
+                          'five_minute_utilization',
+                          'fifteen_minute_utilization', 'mean_utilization'],
+                     snapshot_keys=['median', 'percentile_95th',
+                                    'percentile_99th', 'percentile_999th'])
             if isinstance(metric, Timer):
-                self.send_metric(name, 'timer', metric, 'mean')
+                send(mtype='timer',
+                     keys=['count', 'total_time', 'one_minute_rate',
+                          'five_minute_rate', 'fifteen_minute_rate',
+                          'mean_rate', 'min', 'max', 'mean', 'stddev'],
+                     snapshot_keys=['median', 'percentile_95th',
+                                    'percentile_99th', 'percentile_999th'])
             if isinstance(metric, Counter):
-                self.send_metric(name, 'counter', metric, 'count')
+                send(mtype='counter', keys=['count'])
             if isinstance(metric, Histogram):
-                self.send_metric(name, 'histogram', metric, 'mean')
+                send(mtype='histogram',
+                     keys=['count', 'min', 'max', 'mean', 'stddev'],
+                     snapshot_keys=['median', 'percentile_95th',
+                                    'percentile_99th', 'percentile_999th'])
 
         self._send()
 
-    def send_metric(self, name, mtype, metric, keys):
-        value = getattr(metric, keys, None)
-        if value:
-            self._buffered_send_metric(name, value, mtype)
+    def send_metric(self, name, mtype, metric, keys, snapshot_keys=None):
+        snapshot_keys = snapshot_keys or []
 
-    def _buffered_send_metric(self, name, value, mtype):
         if self.prefix:
-            name = '{0}.{1}'.format(self.prefix, name)
+            name = "{0}.{1}".format(self.prefix, name)
+
+        for name in keys:
```

Comment:
by using `name` in this for loop, you're overwriting `name` variable defined in line 98, so `self.prefix` and `name` of the metric is not used at all (only name of values, like count, rate etc)